modules/navier_stokes/src/kernels/PressureGradient.C

```diff
@@ -20,25 +20,59 @@ PressureGradient::validParams()
   params.addRequiredParam<unsigned int>("component", "number of component (0 = x, 1 = y, 2 = z)");
   params.addClassDescription(
       "Implements the pressure gradient term for one of the Navier Stokes momentum equations.");
+  params.addParam<bool>(
+      "integrate_p_by_parts", true, "Whether to integrate the pressure term by parts");
 
   return params;
 }
 
 PressureGradient::PressureGradient(const InputParameters & parameters)
   : Kernel(parameters),
+    _integrate_p_by_parts(getParam<bool>("integrate_p_by_parts")),
     _component(getParam<unsigned int>("component")),
-    _pressure(coupledValue(NS::pressure))
+    _pressure(coupledValue(NS::pressure)),
+    _grad_pressure(coupledGradient(NS::pressure)),
```

Comment:
MOOSE requests gradients all the time every time regardless of kernel code haha

Reply:
Fair enough.  That's probably not as laughable in the MOOSE context as it would have been for more academic libMesh codes; I'll bet just about every real input file our users run has *some* kernel that needs gradients.